quartz-core/src/main/java/org/quartz/impl/StdSchedulerFactory.java

```diff
@@ -1220,14 +1221,30 @@ else if (schedInstId.equals(SYSTEM_PROPERTY_AS_INSTANCE_ID)) {
     
             if (userTXLocation != null) {
                 UserTransactionHelper.setUserTxLocation(userTXLocation);
+                if (wrapJobInTx) {
```

Comment:
You havent just added some more code here. You have changed the previous behaviour of this class by moving the if(wrapJobInTx) etc inside if(userTXLocation).... is this is a bug fix? If not, how you can be sure that this change does not affect old behaviour?